.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,32 @@
+name: Pull Request
+
+on:
+  pull_request:
+
+jobs:
+  QA:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
+    
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+         
+      - name: Pre commit check
```

Comment:
I think we can skip this one :) out of scope of this PR

Reply:
@qiisziilbash just the pre commit or the whole thing?

Reply:
well that is what blocking the PR for now; I think the goal is minimal addition to publish to pypi; I am not the owner of this repo; so sticking to minimal would be the best